deployment/README.md

```diff
@@ -82,3 +83,88 @@ Click 'Create queue' again and follow the same steps to create a second queue na
+and an API Gateway.
+
+CI will deploy updates to existing Tilegarden instances automatically, but there are
+a few extra steps you'll need to take in order to stand up an entirely new
+instance of Tilegarden. Because Terraform expects a Tilegarden instance to
+exist before it creates infrastructure, **it's best to finish these steps before
+you deploy a new stack with Terraform**.
+
+### 1. Create a remote state bucket for Claudia
+
+Remote state for Claudia should exist in the same S3 bucket as Terraform remote
+state. For staging, this bucket should be something like `staging-pfb-config-us-east-1`.
+In that bucket, create a new folder (that is, an object prefix) for
+Tilegarden remote state called `tilegarden/`.
+
+### 2. Configure environment variables for your function
+
+Tilegarden uses an `.env` file in order to pass environment variables into
+the Lambda function. Copy over the example file to create a new one:
+
+```
+$ cp ./src/tilegarden/.env.example ./src/tilegarden/.env
+```
+
+The three required variables are `AWS_PROFILE`, `PROJECT_NAME`, and
+`LAMBDA_REGION`. Other optional variables can be uncommented and edited to
+reflect your configuration. If you need Tilegarden to access a database, for
+example, you'll likely want to set `LAMBDA_SUBNETS` and `LAMBDA_SECURITY_GROUPS` to point
+to the relevant resources in your VPC.
+
+If you have any additional environment variables (like database connection
+strings) that you need access to in your function, add them to `./src/tilegarden/.env`
+and they will be passed in during deployment.
+
+### 3. Deploy a new Tilegarden instance with Claudia
+
+Before deploying your instance, `touch` a file that Claudia can use to save
+deployment metadata:
+
+```
+$ touch ./src/tilegarden/claudia.json
+```
+
+Next, use the Tilegarden Node scripts in the VM to deploy a new Tilegarden instance:
+
+```
+vagrant@pfb-network-connectivity:/vagrant$ docker-compose \
+                                             -f docker-compose.yml \
+                                             -f docker-compose.test.yml \
+                                             run --rm --entrypoint yarn \
+                                             tilegarden deploy-new
+```
+
+### 4. Update remote state for Claudia and Terraform
+
+Once the deployment has completed, upload your `.env` file and Claudia metadata
+file to the remote state bucket so that CI can update Tilegarden automatically:
+
+```
+$ aws s3 cp ./src/tilegarden/.env s3://<remote-state-bucket>/tilegarden/.env
+$ aws s3 cp ./src/tilegarden/claudia.json s3://<remote-state-bucket>/tilegarden/claudia.json
+```
+
+In addition, edit the remote Terraform variables in `terraform.tfvars` for your
+deployment to point to the domain name for the new API Gateway that Claudia has created.
+The variable for the API Gateway should look something like this:
+
+```hcl
+# Tilegarden
+tilegarden_api_gateway_domain_name = "<api.id>.execute-api.<lambda.region>.amazonaws.com"
```

Comment:
That's a nice pattern! We could do something similar and move this out to `scripts/infra` if we were to use `jq` to parse `claudia.json` and construct the URL. Not really sure if there's a huge amount of benefit since the hostname should never change while the API Gateway still exists -- just would eliminate half of a setup step for standing up a new Tilegarden instance (the step requiring updating the Terraform variable). Is that worth doing in this PR?

Reply:
Your call. I like the idea of minimizing step steps, but agree that it probably won't change much.

Reply:
Great! I think I'll leave it as-is for now, if only to minimize the amount of CI testing we'll have to do to bring this PR in. But I'll keep this pattern in mind for the future.